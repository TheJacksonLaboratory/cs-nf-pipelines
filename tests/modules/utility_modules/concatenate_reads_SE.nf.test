nextflow_process {

    name "Test Process CONCATENATE_READS_SE"
    script "modules/utility_modules/concatenate_reads_SE.nf"
    process "CONCATENATE_READS_SE"

    test("Single-End Data") {

        when {
            params {
                pipeline = 'pta'
                pubdir = "tests/results"
                read_type = "PE"
            }
            process {
                """
                input[0] = tuple('testSample_42', 
                [file('https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/individual_lanes/mm10-WGS-sample_GT42-42424_TTTTTTTT-TTTTTTTT_S42_L001_R1.fastq.gz', checkIfExists: true), file('https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/individual_lanes/mm10-WGS-sample_GT42-42424_TTTTTTTT-TTTTTTTT_S42_L002_R1.fastq.gz', checkIfExists: true)]
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
