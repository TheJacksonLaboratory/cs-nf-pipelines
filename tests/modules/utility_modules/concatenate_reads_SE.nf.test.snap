{
    "Single-End Data": {
        "content": [
            {
                "0": [
                    [
                        "testSample_42",
                        "testSample_42_R1:md5,7ac1324cfd90d6780baea1c1df20e057"
                    ]
                ],
                "concat_fastq": [
                    [
                        "testSample_42",
                        "testSample_42_R1:md5,7ac1324cfd90d6780baea1c1df20e057"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-12T14:04:55.608506131"
    }
}